apps/platform/src/providers/email/RESTEmailProvider.ts

```diff
@@ -0,0 +1,94 @@
+import nodemailer from 'nodemailer'
+import Provider, { ExternalProviderParams, ProviderControllers, ProviderSchema, ProviderSetupMeta } from '../Provider'
+import { createController } from '../ProviderService'
+import EmailProvider from './EmailProvider'
+import { RestTransport } from '@sourceforgery/nodemailer-rest-transport'
+import { decodeHashid, encodeHashid } from '../../utilities'
+import { getUserFromEmail } from '../../users/UserRepository'
+import { getCampaign } from '../../campaigns/CampaignService'
+import { trackMessageEvent } from '../../render/LinkService'
+import Router = require('@koa/router')
+import App from '../../app'
+
+export interface RESTDataParams {
+    host: string
+    port: number
+    secure: boolean
+}
+
+type RESTEmailProviderParams = Pick<RESTEmailProvider, keyof ExternalProviderParams>
+
+export default class RESTEmailProvider extends EmailProvider {
+    host!: string
+    port!: number
+    secure!: boolean
+
+    // declare data: JSONTransport
+
+    static namespace = 'rest'
+    static meta = {
+        name: 'Generic REST',
+        icon: 'https://parcelvoy.com/providers/webhook.svg',
+        paths: {
+            'Webhook URL': `/${this.namespace}`,
+        },
+    }
+
+    static schema = ProviderSchema<RESTEmailProviderParams, RESTDataParams>('restProviderParams', {
+        type: 'object',
+        required: ['host', 'port', 'secure'],
+        properties: {
+            host: { type: 'string' },
+            port: { type: 'number' },
+            secure: { type: 'boolean' },
+        },
+        additionalProperties: false,
+    })
+
+    loadSetup(app: App): ProviderSetupMeta[] {
+        return [{
+            name: 'Webhook URL',
+            value: `http://api:${app.env.port}/api/providers/${encodeHashid(this.id)}/${(this.constructor as any).namespace}`,
+        }]
+    }
+
+    boot() {
+        this.transport = nodemailer.createTransport(
+            new RestTransport({
+                host: this.host,
+                port: this.port,
+                auth: {
+                    api_key: '',
```

Comment:
Why is this an empty string? Shouldn't this be used somewhere? It looks like you do the same thing in the NPM package you created. If there is no security at all this shouldn't exist.

Reply:
Thanks for the feedback!

I have removed the API key and renamed it from Rest to Tachikoma.